src/rust/engine/src/rule_graph.rs

```diff
@@ -3,11 +3,15 @@
 
 use core::{Key, TypeConstraint, TypeId};
 
+use externs::Externs;
+
 use selectors::{Select, Selector, Task};
 
 use std::collections::HashMap;
 use std::collections::HashSet;
 use std::collections::VecDeque;
+use std::collections::hash_map::Entry as StdEntry;
```

Comment:
I think that in general it would be good to prefer relative imports to renaming of things... IE, collapse the following four imports into:

    use std::collections::{hash_map, HashMap, HashSet, VecDeque};


...and then use as:

    hash_map::Entry::Occupied()